homeassistant/helpers/config_validation.py

```diff
@@ -1769,6 +1770,29 @@ def STATE_CONDITION_SCHEMA(value: Any) -> dict[str, Any]:
     )
 )
 
+
+def _backward_compat_trigger_schema(value: Any | None) -> Any:
+    """Rewrite trigger `trigger` to `platform`.
+
+    `platform` has been renamed to `trigger` in user documentation and in the automation
+    editor. The Python trigger implementation still uses `platform`, so we need to
+    rename `trigger` to `platform.
+    """
+
+    if not isinstance(value, Mapping):
+        return value
```

Comment:
Let's explain why we do this instead of fail

```suggestion
        # If the value is not a mapping, we let that be handled by the TRIGGER_SCHEMA
        return value
```